rio_tiler/io/xarray.py

```diff
@@ -71,7 +82,20 @@ def __attrs_post_init__(self):
         assert rioxarray is not None, "rioxarray must be installed to use XarrayReader"
 
         self.bounds = tuple(self.input.rio.bounds())
+
+        if not self.input.rio.crs and default_to_wgs84_crs:
+            warnings.warn(
+                "Dataset doesn't have a valid CRS set. Automatically setting CRS to WGS84",
+                OverwritingCRSWarning,
+            )
+            self.input.rio.write_crs("epsg:4326", inplace=True)
+
         self.crs = self.input.rio.crs
+        if not self.crs:
+            warnings.warn(
+                "Dataset doesn't have a valid CRS set. rio-tiler might not work properly",
+                MissingCRSWarning,
+            )
```

Comment:
Why do we not throw an exception in this case?

Using an `XarrayTiler` for anything meaningful without a (valid) CRS, will inevitably fail.

Reply:
We usually don't fail for this because you can still access the dataset 🤷 

Let me think about it